src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -1376,29 +1659,82 @@ def kubernetes_policy(self) -> dict:
             "kms:DescribeKey"
         ])])
 
+    def _setup_iam_ec2_role(self, local_role_name: str, policies: Dict[str, Any]) -> str:
+        """
+        Create an IAM role with the given policies, using the given name in
+        addition to the cluster name, and return its full name.
+        """
+
+        # Grab the connection we need to use for this operation.
+        iam = self.aws.boto2(self._region, 'iam')
+
+        # Make sure we can tell our roles apart from roles for other clusters
+        aws_role_name = self._namespace_name(local_role_name)
+        try:
+            # Make the role
+            logger.debug('Creating IAM role...')
+            iam.create_role(aws_role_name, assume_role_policy_document=json.dumps({
+                "Version": "2012-10-17",
+                "Statement": [{
+                    "Effect": "Allow",
+                    "Principal": {"Service": ["ec2.amazonaws.com"]},
+                    "Action": ["sts:AssumeRole"]}
+                ]}))
+            logger.debug('Created new IAM role')
+        except BotoServerError as e:
+            if e.status == 409 and e.error_code == 'EntityAlreadyExists':
+                logger.debug('IAM role already exists. Reusing.')
+                pass
+            else:
+                raise
+
+        # Delete superfluous policies
+        policy_names = set(iam.list_role_policies(aws_role_name).policy_names)
+        for policy_name in policy_names.difference(set(list(policies.keys()))):
+            iam.delete_role_policy(aws_role_name, policy_name)
```

Comment:
Does this delete only duplicate policies or does it delete potentially policies it needs?  I'm only concerned that this might interfere with a user supplied policy a la `--awsEc2ProfileArn` if it is the second: https://github.com/DataBiosphere/toil/pull/2703

Reply:
If someone specifies a --awsEc2ProfileArn, `_createProfileArn()` is never called, and it won't call `_setup_iam_ec2_role()`. This is supposed to run just on properly-cluster-namespaced, toil-managed roles, where Toil is allowed to fiddle with all the policies.

Now, I copied most of this over from the old Boto2Context, which used to be CGCloud and had ambitions of idempotency, so it has all this code to do things like fix up an existing role despite how you shouldn't ever `launch-cluster` with the same name as an existing cluster. We might want to simplify it away and just bail out if the role already exists.